src/main/java/org/springframework/data/redis/connection/ReactiveStreamCommands.java

```diff
@@ -337,7 +337,7 @@ public Long getMaxlen() {
 		 * @since 2.3
 		 */
 		public boolean hasMaxlen() {
-			return maxlen != null && maxlen > 0;
+			return maxlen != null;
```

Comment:
For context, Lettuce and Jedis treat `maxlen` edge case differently. `XAddArgs` in Lettuce asserts `maxlen > 0` whereas Jedis doesn't check on `maxlen` so it follows default redis behaviour which asserts `maxlen >=0`.  My best choice was just let the user know. So I didn't put any assertion on `maxlen`.

```
java.lang.IllegalArgumentException: Maxlen must be greater 0

	at io.lettuce.core.internal.LettuceAssert.isTrue(LettuceAssert.java:193)
	at io.lettuce.core.XAddArgs.maxlen(XAddArgs.java:121)
	at org.springframework.data.redis.connection.lettuce.LettuceStreamCommands.xAdd(LettuceStreamCommands.java:84)
	at org.springframework.data.redis.connection.DefaultedRedisConnection.xAdd(DefaultedRedisConnection.java:489)
	at org.springframework.data.redis.connection.DefaultStringRedisConnection.xAdd(DefaultStringRedisConnection.java:2918)
	at org.springframework.data.redis.core.DefaultStreamOperations.lambda$add$2(DefaultStreamOperations.java:153)
```